apt_builder/generate_dockerfile.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2017 Google Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"github.com/ghodss/yaml"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+)
+
+var INTERMEDIATE_IMAGE_TAG = "apt:intermediate"
+
+func generateDockerfile(installer Installer) (int, string) {
+	var err error
+
+	build_dir, err := ioutil.TempDir(".", "intermediate_base.")
+	if err != nil {
+		log.Printf("Error when creating temp_dir: %s", err)
+		return 1, ""
+	}
+
+	f, err := os.Create(build_dir + "/Dockerfile")
+	if err != nil {
+		log.Printf("Error opening file for writing: %s", err)
+		return 1, build_dir
+	}
+
+	defer f.Close()
+
+	w := bufio.NewWriter(f)
+
+	err = INSTALL_TMPL.Execute(w, installer)
+	if err != nil {
+		log.Printf("Error when executing template: %s", err)
+		return 1, build_dir
+	}
+	for _, ppa := range installer.AptPackages.PPAs {
```

Comment:
I think you might be able to do all of this in one template. You can do loops with the "range" operator.